### ISSUE TITLE
Better way to register APIs

### ISSUE THREAD

**Issue by uttamk** [Aplopio/rip#9]
Today the way to register APIs is to give it a leading name and version number which is kind of hardcoded.
```python
rbox_api_v2 = Api(name='api', version='v1')
```
This translates to
api/v1/xxx
Both name and version should be optional. 
Ideally it should be root accessible / without a version number

**Comment by raghuveerkancherla** [comment]
api = Api(name='api/v1')

**Comment by kracekumar** [comment]
+1 no version number.

api = Api()
api.register('users', MyUserResource)

...
So People in flask want to use decorators, so api should be flexible

@api.route('/users')
class MyUserResource:
....

or

@api.route('/users')
def user_resource():
.....



On Mon, Jan 5, 2015 at 10:40 AM, Raghu <notifications@github.com> wrote:

> api = Api(name='api/v1')
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Aplopio/rip/issues/9#issuecomment-68669098>.
>



-- 
Regards
Kracekumar - +91-85530 29521
"Talk is cheap, show me the code" - Linus Torvalds

**Comment by raghuveerkancherla** [comment]
I think we should keep the url routing part separate from the resources.
Resources implement CRUD without any assumptions about the delivery
mechanism (http). the decorator suggestion violates that.

**Comment by kracekumar** [comment]
Not sure. How current registration doesn't violate it.

api = Api(name='api', version='v1')
api.register_resource('foo', FooResource())

And decorator can be written as

@register(api_object, 'resource') as well

If you are using http, at some point bridge needs to exist. Calling a
method is one way and decorator is another way.




On Mon, Jan 5, 2015 at 1:42 PM, Raghu <notifications@github.com> wrote:

> I think we should keep the url routing part separate from the resources.
> Resources implement CRUD without any assumptions about the delivery
> mechanism (http). the decorator suggestion violates that.
>
> On Mon, Jan 5, 2015 at 1:37 PM, Kracekumar Ramaraju <
> notifications@github.com> wrote:
>
> > +1 no version number.
> >
> > api = Api()
> > api.register('users', MyUserResource)
> >
> > ...
> > So People in flask want to use decorators, so api should be flexible
> >
> > @api.route('/users')
> > class MyUserResource:
> > ....
> >
> > or
> >
> > @api.route('/users')
> > def user_resource():
> > .....
> >
> >
> >
> > On Mon, Jan 5, 2015 at 10:40 AM, Raghu <notifications@github.com>
> wrote:
> >
> > > api = Api(name='api/v1')
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/Aplopio/rip/issues/9#issuecomment-68669098>.
> > >
> >
> >
> >
> > --
> > Regards
> > Kracekumar - +91-85530 29521
> > "Talk is cheap, show me the code" - Linus Torvalds
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/Aplopio/rip/issues/9#issuecomment-68678209>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Aplopio/rip/issues/9#issuecomment-68678534>.
>



-- 
Regards
Kracekumar - +91-85530 29521
"Talk is cheap, show me the code" - Linus Torvalds

**Comment by raghuveerkancherla** [comment]
Yes but register_resource is at http_api level not at resource level.
Decorator on the other hand will have to be at Resource level
